db/migrate/20190208233321_create_visualizations.rb

```diff
@@ -0,0 +1,19 @@
+class CreateVisualizations < ActiveRecord::Migration[5.1]
+  def change
+    create_table :visualizations do |t|
+      t.references :user, foreign_key: true
```

Comment:
In general, we have been avoiding foreign keys. We removed all of them a while ago because of database locking issues and are now just enforcing them in the code where necessary. I’d recommend the same here

Reply:
ok good to know!